azure-pipelines.yml

```diff
@@ -16,14 +16,54 @@ jobs:
   steps:
   - script: |
       printenv
-        echo hello
-        pwd
-        ls
+      pwd
+      ls
+      echo "agend build directory"
+      echo $(Agent.BuildDirectory)
     displayName: filesystem check
   - script: |
+      cd $(Agent.BuildDirectory)/s
+      git clone https://github.com/oci-labs/rmw_opendds.git
+    displayName: clone rmw_opendds
+  - script: |
+        cd $(Agent.BuildDirectory)
+        git clone https://github.com/adamsj-oci/rmw_build
+        vcs import s < rmw_build/ros2.repos
+        echo "ls"
+        ls
+        echo "ls s"
+        ls s/
+    displayName:  rmw build repo and dependencies
+  - script: |
+        cd $(Agent.BuildDirectory)
+        source /opt/OpenDDS/setenv.sh
+        source /opt/ros/eloquent/setup.bash
+        printenv|grep DDS
+        printenv|grep ROS
+        colcon build --symlink-install --packages-up-to rmw_opendds_cpp
+    displayName:  build up to rmw_opendds_cpp
+  - script: |
+        cd $(Agent.BuildDirectory)
+        source /opt/OpenDDS/setenv.sh
+        source /opt/ros/eloquent/setup.bash
+        source install/local_setup.bash
+        printenv|grep DDS
+        printenv|grep ROS
+        colcon build --symlink-install --packages-up-to rcl_interfaces examples_rclcpp_minimal_publisher examples_rclcpp_minimal_subscriber
+    displayName:  build simple tests
+  - script: |
+        cd $(Agent.BuildDirectory)
         source /opt/OpenDDS/setenv.sh
         source /opt/ros/eloquent/setup.bash
+        source install/local_setup.bash
         printenv|grep DDS
         printenv|grep ROS
-        colcon build --symlink-install
-    displayName:  environment setup and build
+        export RMW_IMPLEMENTATION=rmw_opendds_cpp
+        env|grep RMW
+        ros2 run examples_rclcpp_minimal_subscriber subscriber_member_function_twenty 1> sub.out& 
+        process_id=$!
+        ros2 run examples_rclcpp_minimal_publisher publisher_member_function_twenty
+        wait $process_id
+        cat sub.out
+        grep "I heard" sub.out
```

Comment:
Probably not really part of this PR since we're just looking for clean build and not expecting a test to successfully run. We should be thinking about the next step for a runtime test for services if that means the minimal examples or getting the system tests to run.

Reply:
I assumed this was not a PR specific change, but more a long term, general CI test setup. I am OK removing pub / sub tests though, if you think it does not belong to this package - can you please confirm?

Reply:
I think pub/sub does belong here. It tests the other feature pub/sub that we want to preserve. Eventually we may want to add the client/service tests but only when it's closer to working with OpenDDS.